server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/ElasticsearchClient.java

```diff
@@ -108,6 +109,23 @@ public ElasticsearchClient() {
     }
 
     /**
+     * This method explicit instantiates the singleton. For use outside a CDI environment.
+     *
+     * @return The singleton
+     */
+    public static synchronized ElasticsearchClient getSingleton() {
+        if (singleton == null) {
+            singleton = new ElasticsearchClient();
+            try {
+                singleton.init();
```

Comment:
if init() is synchronous, it's fine.
I had an error in a similar assumption for a Cassandra driver, where the init() operation was asynchronous and then the getSingleton() could return a instance without fully initialization (and make the debug more difficult).
Just a minor note to avoid headaches.